### ISSUE TITLE
Ulimit should use Long instead of Integer

### ISSUE THREAD

**Issue by oobles** [spotify/docker-client#550]
## Description

HostConfig Ulimit definition uses Integer.  This should be changed to Long.

## How to reproduce

Create a container with --ulimit memlock=17179869184  in command line.  ie memlock value in bytes greater than integer max.

Get host config via api.

## What do you expect

Not getting an out of range exception.

## What happened instead

com.fasterxml.jackson.databind.JsonMappingException: Numeric value (17179869184) out of range of int
at [Source: org.glassfish.jersey.message.internal.ReaderInterceptorExecutor$UnCloseableInputStream@50d6580c; line: 1, column: 2312] (through reference chain: com.spotify.docker.client.messages.ContainerInfo["HostConfig"]->com.spotify.docker.client.messages.HostConfig["Ulimits"]->com.spotify.docker.client.messages.Ulimit["Hard"])

## Software:

- `docker version`:  1.12.3
- Spotify's docker-client version: 6.0.0

## Full backtrace

```text
com.fasterxml.jackson.databind.JsonMappingException: Numeric value (17179869184) out of range of int
at [Source: org.glassfish.jersey.message.internal.ReaderInterceptorExecutor$UnCloseableInputStream@50d6580c; line: 1, column: 2312] (through reference chain: com.spotify.docker.client.messages.ContainerInfo["HostConfig"]->com.spotify.docker.client.messages.HostConfig["Ulimits"]->com.spotify.docker.client.messages.Ulimit["Hard"])
       at com.fasterxml.jackson.databind.JsonMappingException.wrapWithPath(JsonMappingException.java:210)
       at com.fasterxml.jackson.databind.JsonMappingException.wrapWithPath(JsonMappingException.java:177)
       at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.wrapAndThrow(BeanDeserializerBase.java:1475)
       at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:258)
       at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:125)
       at com.fasterxml.jackson.datatype.guava.deser.GuavaImmutableCollectionDeserializer._deserializeContents(GuavaImmutableCollectionDeserializer.java:42)
       at com.fasterxml.jackson.datatype.guava.deser.GuavaImmutableCollectionDeserializer._deserializeContents(GuavaImmutableCollectionDeserializer.java:14)
       at com.fasterxml.jackson.datatype.guava.deser.GuavaCollectionDeserializer.deserialize(GuavaCollectionDeserializer.java:103)
       at com.fasterxml.jackson.databind.deser.SettableBeanProperty.deserialize(SettableBeanProperty.java:520)
       at com.fasterxml.jackson.databind.deser.impl.FieldProperty.deserializeAndSet(FieldProperty.java:101)
       at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:256)
       at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:125)
       at com.fasterxml.jackson.databind.deser.SettableBeanProperty.deserialize(SettableBeanProperty.java:520)
       at com.fasterxml.jackson.databind.deser.impl.FieldProperty.deserializeAndSet(FieldProperty.java:101)
       at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:256)
       at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:125)
       at com.fasterxml.jackson.databind.ObjectReader._bind(ObjectReader.java:1470)
       at com.fasterxml.jackson.databind.ObjectReader.readValue(ObjectReader.java:912)
       at com.fasterxml.jackson.jaxrs.base.ProviderBase.readFrom(ProviderBase.java:811)
       at org.glassfish.jersey.message.internal.ReaderInterceptorExecutor$TerminalReaderInterceptor.invokeReadFrom(ReaderInterceptorExecutor.java:266)
       at org.glassfish.jersey.message.internal.ReaderInterceptorExecutor$TerminalReaderInterceptor.aroundReadFrom(ReaderInterceptorExecutor.java:236)
       at org.glassfish.jersey.message.internal.ReaderInterceptorExecutor.proceed(ReaderInterceptorExecutor.java:156)
       at org.glassfish.jersey.message.internal.MessageBodyFactory.readFrom(MessageBodyFactory.java:1085)
       at org.glassfish.jersey.message.internal.InboundMessageContext.readEntity(InboundMessageContext.java:853)
```

**Comment by oobles** [comment]
We patched 6.0.0 successfully with the following changes.

   public static class Ulimit {
        @JsonProperty("Name")
        private String name;
        @JsonProperty("Soft")
        private Long soft;
        @JsonProperty("Hard")
        private Long hard;

        public Ulimit() {
        }

        private Ulimit(final Builder builder) {
            this.name = builder.name;
            this.soft = builder.soft;
            this.hard = builder.hard;
        }

        public static Builder builder() {
            return new Builder();
        }

        public String name() {
            return name;
        }

        public Long soft() {
            return soft;
        }

        public Long hard() {
            return hard;
        }

        @Override
        public boolean equals(final Object o) {
            if (this == o) {
                return true;
            }
            if (o == null || getClass() != o.getClass()) {
                return false;
            }

            final Ulimit that = (Ulimit) o;

            return Objects.equals(this.name, that.name) &&
                    Objects.equals(this.soft, that.soft) &&
                    Objects.equals(this.hard, that.hard);
        }

        @Override
        public int hashCode() {
            return Objects.hash(name, soft, hard);
        }

        @Override
        public String toString() {
            return MoreObjects.toStringHelper(this)
                    .add("name", name)
                    .add("soft", soft)
                    .add("hard", hard)
                    .toString();
        }

        public static class Builder {
            private String name;
            private Long soft;
            private Long hard;

            private Builder() {
            }

            public Ulimit build() {
                return new Ulimit(this);
            }

            public Builder name(final String name) {
                this.name = name;
                return this;
            }

            public Builder soft(final Long soft) {
                this.soft = soft;
                return this;
            }

            public Builder hard(final Long hard) {
                this.hard = hard;
                return this;
            }
        }
    }

**Comment by mattnworb** [comment]
@oobles want to send your changes as a PR? I'd be happy to merge
